src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs

```diff
@@ -76,6 +77,11 @@ public sealed class AveragedPerceptronTrainer : AveragedLinearTrainer<BinaryPred
 
         private readonly Options _args;
 
+        internal class AveragedPerceptronDefault : AveragedDefault
```

Comment:
>AveragedPerceptronDefault [](start = 23, length = 25)

I am wondering is it worth it to introduce a new class just to contain a const value?

Reply:
The pattern so far in ML.NET has been to have classes to contain the default values, so I have just continued that pattern. While it does seem weird to have a class with just 1 value, I think keeping the pattern so that the code is consistent is better then breaking with consistency. Thoughts?

Reply:
I still not familiar with the whole space so I cannot judge that. but in general I am seeing ending up with many classes would make it difficult to use I guess. please don't block on my comment. I am just thinking loudly with you.

Reply:
can we put this constant in AveragedDefault and rename it to something like PerceptronNumberOfIterations or something similar?

Reply:
I am not sure how well that will look. If you look [here](https://github.com/dotnet/machinelearning/blob/c2ddff17abbaf4d873ae3a805f2688d4a11f7988/src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineLinear.cs#L29), we have a bunch of attributes that affect how it shows up with the command line interface. Since we are using a class and overriding just that value, those attributes (I think) will come as well. I am not sure how it will behave if we have a constant value with a different name we use.

Reply:
got it. please continue with your original plan if no one else have more thoughts about it.